pymeasure/instruments/hp/hp3478A.py

```diff
@@ -0,0 +1,489 @@
+                self.ERR_cal, self.ERR_RAM, self.ERR_ROM, self.ERR_slope, self.ERR_AD, self.ERR_AD_Link,
+                self.DAC_value)
+            )
+
+class Status(ctypes.Union):
+    """Union type element for the decoding of the status bit-fields
+    """
+    _fields_ = [
+        ("B", Status_bytes),
+        ("b", Status_bits)
+    ]
+
+
+
+class HP3478A(Instrument):
+    """ Represents the Hewlett Packard 3748A 5 1/2 digit multimeter
+    and provides a high-level interface for interacting
+    with the instrument.
+
+    As this unit predates SCPI some tricks are required to get this working
+
+    """
+
+    def __init__(self, resourceName, **kwargs):
+        super(HP3478A, self).__init__(
+            resourceName,
+            "Hewlett-Packard HP3478A",
+            includeSCPI = False,
+            send_end = True,
+            read_termination = "\r\n",
+            **kwargs
+        )
+
+    # Definitions for different specifics of this instrument
+    MODES={"DCV": "F1",
+           "ACV": "F2",
+           "R2W": "F3",
+           "R4W": "F4",
+           "DCI": "F5",
+           "ACI": "F6",
+           "Rext": "F7",
+           }
+
+    RANGES={"DCV": {3E-2: "R-2", 3E-1: "R-1",3: "R0",30: "R1",300:"R2",
+                    "auto": "RA"},
+            "ACV": {3E-1: "R-1", 3: "R0",30: "R1" ,300: "R2","auto": "RA"},
+            "R2W": {30: "R1", 300: "R2", 3E3: "R3",3E4: "R4",3E5: "R5",
+                    3E6: "R6", 3E7: "R7","auto": "RA"},
+            "R4W": {30: "R1",300: "R2", 3E3: "R3",3E4: "R4",3E5: "R5",
+                    3E6: "R6", 3E7: "R7","auto": "RA"},
+            "DCI": {3E-1: "R-1",3: "R0","auto": "RA"},
+            "ACI": {3E-1: "R-1",3: "R0","auto": "RA"},
+            "Rext": {3E7: "R7","auto": "RA"},
+            }
+
+    TRIGGERS={
+            "auto": "T1",
+            "internal": "T1",
+            "external": "T2",
+            "single": "T3",
+            "hold": "T4",
+            "fast": "T5",
+            }
+
+    ERRORS={
+            1: "Calibration error",
+            2: "RAM error",
+            4: "ROM error",
+            8: "AD slope error",
+            16: "AD converter error",
+            32: "AD link error",
+           }
+    
+
+
+    #decoder functions
+    @classmethod
+    def decode_status(self,status_bytes, field=None):
+        """Method to handle the decoding of the status bytes into something meaningfull
+
+        :param status_bytes:    list of bytes to be decoded
+        :return ret_val:
+
+        """
+        ret_val = Status(Status_bytes(*status_bytes))
+        if field is None:
+            return ret_val.b
+        elif field == "SRQ":
+            return getattr(ret_val.B, "byte3")
+        else:
+            return getattr(ret_val.b, field)
+
+    @classmethod
+    def decode_mode(cls,status_bytes):
+        """Method to decode current mode
+
```

Comment:
If you replace `@classmethod` with `@staticmethod` and remove the `cls` parameter, you should be good to go with static method.

Reply:
Hmm, I think I'd actually suggest leaving this (and `decode_range`) a classmethod and changing `HP3478A.MODES` to `cls.MODES` (and also `HP3478A.RANGES` to `cls.RANGES` in `decode_range`); or is there a reason not to do that?

(`decode_status` and `decode_trigger` can indeed be converted to a staticmethod anyway)

Reply:
@CasperSchippers I think you are right, I didn't notice this. Sorry for the confusion

Reply:
Thanks for the contious support.
I will chage in decode_mode & decode_range the HP3478A. references to cls.
Also I will change decode_status and decode_trigger to static methods.